lib/nmatrix/lapack.rb

```diff
@@ -1,8 +1,20 @@
 module NumRuby::Linalg
-  def self.inv(obj)
-    if obj.is_a?(NMatrix)
-      return obj.invert
+  def self.inv(matrix)
+    if not matrix.is_a?(NMatrix)
+      raise("Invalid matrix. Not of type NMatrix.")
+    end
+    if matrix.dim != 2
+      raise("Invalid shape of matrix. Should be 2.")
+    end
+    if matrix.shape[0] != matrix.shape[1]
+      raise("Invalid shape. Expected square matrix.")
     end
+    m, n = matrix.shape
```

Comment:
what do you main by `m` and `n`?

Reply:
`m` is rows count and `n` is columns count.

Reply:
I think I could name these fields as `row` and` account_ columns` to make this more readable. WDYT? 🤔